### PR TITLE
Switched to github.com/shirou/gopsutil/v3

### PR DESCRIPTION
**Pull request by iglendd**
### What does this PR do?
Switched to github.com/shirou/gopsutil/v3

The change was relatively straightforward. In addition to go.mod changes replaced all import of `shirou/gopsutil[/...]`  on `shirou/gopsutil/v3[/...]`. Only in few files we had to change source code to reflect v3 API changes. It worth to note that in preliminary step we also changed DataDog/gohai to use `shirou/gopsutil/v3` as well. There are no more indirect or incompatible _shirou/gopsutil_ references in go.mod.

### Motivation
Support for shirou/gopsutil/v2 is ended

### Additional Notes
n/a

### Possible Drawbacks / Trade-offs
n/a

### Describe how to test/QA your changes
No special testing is needed. Passing automated tests is sufficient.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by albertvaka — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

File: pkg/collector/corechecks/system/disk/iostats.go
Comment (djmitche) [inline pkg/collector/corechecks/system/disk/iostats.go:21]: Is this still not working?

**Review by djmitche — APPROVED** [review]

File: pkg/collector/corechecks/system/disk/iostats.go
Comment (iglendd) [reply pkg/collector/corechecks/system/disk/iostats.go:21]: @djmitche Thank you. No idea if it is working. I was replacing all references to implicit import to explicit. I just changed a comment as well. Not sure if it was correct thing to do but I think it was.

Comment (djmitche) [reply pkg/collector/corechecks/system/disk/iostats.go:21]: It looks like it's _not_ exported, so maybe update the comment?

Comment (iglendd) [reply pkg/collector/corechecks/system/disk/iostats.go:21]: @djmitche and @albertvaka what should I do about it. He is right the comment is not correct anymore. The const is not exported in v3. It does not break anything and it is only inaccurate comment with the same consequences. The problem that the PR is is merged already. Should I create backlog jira to change the comment alone?

Comment (djmitche) [reply pkg/collector/corechecks/system/disk/iostats.go:21]: It doesn't look like it's merged..

image

But if that's incorrect, then just make another PR to change the comment - no need for a backlog card, sprint planning, etc.

Comment (iglendd) [reply pkg/collector/corechecks/system/disk/iostats.go:21]: You are right! Damn. it was a while ago I was waiting for everyone's approval and shifted my focus on other issue. I thought that that I have merged it already. Will do it soon. Thank you again.

Comment (iglendd) [reply pkg/collector/corechecks/system/disk/iostats.go]: Please see my last commit which should address your comment. Thank you.

Comment (iglendd) [reply pkg/collector/corechecks/system/disk/iostats.go]: Strange, change a comment and now lot of things are failing :(

Comment (iglendd) [reply pkg/collector/corechecks/system/disk/iostats.go]: ... plus conflicts

**Review by djmitche — APPROVED** [review]

**Review by gbbr — APPROVED** [review]

**Pull request merged**
